clang/test/SemaSYCL/stdtypes_kernel_type.cpp

```diff
@@ -8,36 +8,55 @@ typedef long int ptrdiff_t;
 typedef decltype(nullptr) nullptr_t;
 class T;
 class U;
+class Foo;
 } // namespace std
 
 template <typename T>
 struct Templated_kernel_name;
 
+template <typename T>
+struct Templated_kernel_name2;
+
 template <typename T, typename... Args> class TemplParamPack;
 
 using namespace cl::sycl;
 queue q;
 
 int main() {
 #ifdef CHECK_ERROR
-  // expected-error@Inputs/sycl.hpp:220 5 {{kernel name cannot be a type in the "std" namespace}}
   q.submit([&](handler &h) {
+    // expected-error@Inputs/sycl.hpp:220 {{'nullptr_t' is an invalid kernel name type}}
+    // expected-note@Inputs/sycl.hpp:220 {{type 'nullptr_t' cannot be in the "std" namespace}}
```

Comment:
Could this note be changed to: "type 'nullptr_t' is in the "std" namespace"
Right now as it reads, it looks as if nullptr_t shouldn't be there; but this is not something change-able.

Reply:
Yea I thought about this but I wasn't sure if we need to explicitly tell the users that types used in kernel name cannot be in the `std `namespace. We're kind of implying it when we say "Error: invalid kernel name. Note: type is in std namespace", but we won't be explicitly saying "std types cannot be used in kernel name". Do you think that is good enough though?

Reply:
I just think:
error: "'x' is an invalid kernel name type" (because) note: "'y' is in the "std" namespace"
reads better than:
error: "'x' is an invalid kernel name type" (because) note: "'y' cannot be in the "std" namespace"

This is just based on the two examples I have seen so far.  Are there contexts where the latter reads better?

Reply:
The second doesn't look prettier for me, but it explains the problem with kernel name better, so it is more informative. I don't see a good way to make it pretty and informative without adding some additional note which will say something like "SYCL spec doesn't allow to use types in std namespace in kernel names" but won't it be too long?

Reply:
Not really. I just thought "cannot" made the message clearer, when it came to the reasoning why the kernel name is invalid. I see your point of how it might be misunderstood though. I I'll make the change and upload patch shortly. Thanks for review!

Reply:
@Fznamznon yea I agree an additional note is overkill. I think we just need to decide between option 1 and 2 here. Personally I prefer option 2 but I see why @premanandrao has concerns. It could be read as 'nullptr' can't be in std namespace as opposed to kernel name type cannot be in std namespace. I was hoping the user would be able to understand the intention there.

Reply:
> I think we just need to decide between option 1 and 2 here.

I'm fine with both options, so I will leave final decision up to you and @premanandrao .

Reply:
> I'm fine with both options, so I will leave final decision up to you and @premanandrao .

@elizabethandrews and I just talked offline.  Mine wasn't a strong opinion and since @Fznamznon feels that "cannot be ..." makes it more informative, and I don't have a better suggestion, we decided to leave it as is.